src/dune_engine/compound_user_error.ml

```diff
@@ -1,7 +1,7 @@
 open Import
 
 module T : sig
-  type t = private
+  type t =
```

Comment:
Brought it back.

Reply:
Hmm, there might be some confusion here. A `Compound_user_error.t` is always a single error. The `related` field is just for additional locations the error might have.

> Also, unless there is good reason to think that in the case of multiple errors the first one is especially meaningful, I would just call the fields hd and tl. Or, introduce a generic Nonempty_list.t type in dune if it doesn't already exist, and then make User_errors.t a User_message.t Nonempty_list.t. This can be resolved in a separate PR.

That would be possible. But note that there are some differences between `main` and `related` that aren't reflected adequately in the types. For example, every single "related" message must have a location. While the `main` message may omit it. Should I perhaps improve the type of `related` here instead?